custom/icds_reports/management/commands/run_aggregation_query.py

```diff
@@ -0,0 +1,112 @@
+    _aggregate_child_health_thr_forms, _aggregate_ccs_record_thr_forms, _aggregate_child_health_pnc_forms,
+    _aggregate_ccs_record_pnc_forms, _aggregate_delivery_forms, _aggregate_bp_forms,
+    _aggregate_awc_infra_forms, _child_health_monthly_table, _agg_ls_awc_mgt_form, _agg_ls_vhnd_form,
+    _agg_beneficiary_form, create_mbt_for_month, setup_aggregation, _agg_ls_table,
+    _update_months_table, _daily_attendance_table, _agg_child_health_table,
+    _ccs_record_monthly_table, _agg_ccs_record_table, _agg_awc_table, _agg_awc_table_weekly,
+    aggregate_awc_daily, email_dashboad_team
+)
+
+
+STATE_TASKS = {
+    'aggregate_gm_forms': _aggregate_gm_forms,
+    'aggregate_df_forms': _aggregate_df_forms,
+    'aggregate_cf_forms': _aggregate_cf_forms,
+    'aggregate_ccs_cf_forms': _aggregate_ccs_cf_forms,
+    'aggregate_child_health_thr_forms': _aggregate_child_health_thr_forms,
+    'aggregate_ccs_record_thr_forms': _aggregate_ccs_record_thr_forms,
+    'aggregate_child_health_pnc_forms': _aggregate_child_health_pnc_forms,
+    'aggregate_ccs_record_pnc_forms': _aggregate_ccs_record_pnc_forms,
+    'aggregate_delivery_forms': _aggregate_delivery_forms,
+    'aggregate_bp_forms': _aggregate_bp_forms,
+    'aggregate_awc_infra_forms': _aggregate_awc_infra_forms,
+    'agg_ls_awc_mgt_form': _agg_ls_awc_mgt_form,
+    'agg_ls_vhnd_form': _agg_ls_vhnd_form,
+    'agg_beneficiary_form': _agg_beneficiary_form,
+    'create_mbt_for_month': create_mbt_for_month
+}
+
+ALL_STATES = {
+    'child_health_monthly': _child_health_monthly_table
+}
+
+NORMAL_TASKS = {
+    'setup_aggregation': setup_aggregation,
+    'agg_ls_table': _agg_ls_table,
+    'update_months_table': _update_months_table,
+    'daily_attendance': _daily_attendance_table,
+    'agg_child_health': _agg_child_health_table,
+    'ccs_record_monthly': _ccs_record_monthly_table,
+    'agg_ccs_record': _agg_ccs_record_table,
+    'agg_awc_table': _agg_awc_table,
+    'aggregate_awc_daily': aggregate_awc_daily,
+}
+
+
+@attr.s
+class AggregationQuery(object):
+    by_state = attr.ib()
+    func = attr.ib()
+
+
+class Command(BaseCommand):
+    help = "Run portion of dashboard aggregation. Used by airflow"
+
+    def add_arguments(self, parser):
+        parser.add_argument('query_name')
+        parser.add_argument('date')
+        parser.add_argument('interval', type=int)
+
+    def handle(self, query_name, date, interval, **options):
+        self.function_map = {}
+        self.setup_tasks()
+        agg_date = self.get_agg_date(date, interval)
+        query = self.function_map[query_name]
+        if query.by_state:
+            pool = Pool(10)
+            state_ids = cache.get('agg_state_ids_{}'.format(date))
+            if not state_ids:
+                raise CommandError('Aggregation improperly set up. State ids missing')
+            for state in state_ids:
+                pool.spawn(query.func, state, agg_date)
+            pool.join()
+        elif query.by_state is None:
```

Comment:
Does `query.by_state = None` mean that the state_ids are handled internally? I think you'll want to change that so that the state_ids are handled in the same way as other `by_state`. At the moment child_health_monthly will still depend on celery https://github.com/dimagi/commcare-hq/blob/a89c87b26a44dad77a2228a1ac43b5a7ef0a92b6/custom/icds_reports/tasks.py#L574

Reply:
Yeah, that was a kludge for testing. I can rewrite that task to be handled more like the other ones (and spin off the states with gevent)